log/log.go

```diff
@@ -0,0 +1,332 @@
+		}
+		data["bandwidth_in_"+k] = fmt.Sprintf("%.0f", float64(v.bwIn/1024)/duration)
+		data["bandwidth_out_"+k] = fmt.Sprintf("%.0f", float64(v.bwOut/1024)/duration)
+	}
+
+	c.metrics = &metrics{
+		Requests: map[string]int64{},
+		PerfData: map[string]*PerfData{},
+	}
+
+	for k := range c.metrics.Requests {
+		c.metrics.Requests[k] = 0
+	}
+	for k := range c.metrics.PerfData {
+		c.metrics.PerfData[k] = &PerfData{
+			rtMax:   0,
+			rtMin:   0,
+			rtCount: 0,
+			rt:      0,
+		}
+	}
+	return data, nil
+}
+
+func tagString(tags []string, matches []string) string {
+	res := []string{}
+	for i, tag := range tags {
+		if strings.HasPrefix(tag, "tag_") {
+			res = append(res, matches[i])
+		}
+	}
+	rs := strings.Join(res, ".")
+	if rs != "" {
+		rs = "." + rs
+	}
+	return rs
+}
+
+func index(data []string, key string) int {
+	for i, v := range data {
+		if key == v {
+			return i
+		}
+	}
+	return -1
+}
+
+func (c *collector) tailLogFile(name string, conf *LogTarget) {
+	rTokenize, err := regexp.Compile(conf.Format.Pattern)
+	if err != nil {
+		log.Println("WARN: Fatal error watching file", conf.Path, ":", err)
+		return
+	}
+	numTokens := len(rTokenize.SubexpNames())
+	var rInclude *regexp.Regexp
+	var rExclude *regexp.Regexp
+
+	if conf.Filter.Include != "" {
+		rInclude, err = regexp.Compile(conf.Filter.Include)
+		if err != nil {
+			log.Println("WARN: Fatal error watching file", conf.Path, ":", err)
+			return
+		}
+	}
+	if conf.Filter.Exclude != "" {
+		rExclude, err = regexp.Compile(conf.Filter.Exclude)
+		if err != nil {
+			log.Println("WARN: Fatal error watching file", conf.Path, ":", err)
+			return
+		}
+	}
+
+	for {
+		if _, err := os.Stat(conf.Path); err != nil {
+			log.Println("WARN: Could not watch file", conf.Path, ":", err, ". Retrying in", c.retry)
+			time.Sleep(c.retry * time.Second)
+			continue
+		}
+		t, err := tail.TailFile(conf.Path, tailConfig)
+		if err != nil {
+			log.Println("WARN: Error watching file", conf.Path, ":", err)
+			return
+		}
+
+		for line := range t.Lines {
+			if line.Err != nil {
+				log.Println("WARN: Error watching file", conf.Path, ":", err)
+				continue
+			}
+			if rInclude != nil && !rInclude.MatchString(line.Text) {
+				continue
+			}
+			if rExclude != nil && rExclude.MatchString(line.Text) {
+				continue
+			}
+			s := rTokenize.FindStringSubmatch(line.Text)
+			if len(s) != numTokens {
+				continue
+			}
+			names := rTokenize.SubexpNames()
+			tagString := tagString(names, s)
+
+			codeIdx := index(names, "code")
+			method := index(names, "method")
+			if codeIdx > -1 {
+				reqName := s[method] + "." + s[codeIdx] + "." + name + tagString
+				// NOTE: consider commit at the end to avoid locking N times
+				c.metrics.Lock()
+				if _, ok := c.metrics.Requests[reqName]; ok {
+					c.metrics.Requests[reqName]++
+				} else {
+					c.metrics.Requests[reqName] = 1
+				}
+				c.metrics.Unlock()
+			}
+
+			reqName := s[method] + "_" + name + tagString
```

Comment:
in line 252 a dot is used to prefix the name whereas here an underscore is used. Please use dot everywhere to be consistent (better for regex parsing in prometheus relabelling phase)